backend/app/models/sample_type.rb

```diff
@@ -0,0 +1,284 @@
+# sample_types table
+class SampleType < ActiveRecord::Base
+
+  validates :name,        presence: true, uniqueness: { case_sensitive: false }
+  validates :description, presence: true
+
+  # Return all sample types.
+  #
+  # @return all sample types
+  def self.find_all
+    sql = "
+      select * from sample_types order by name
+    "
+    SampleType.find_by_sql sql
+  end
+
+  def self.find_id(id)
+    sql = "
+      select * from sample_types where id = #{id} limit 1
+    "
+    (SampleType.find_by_sql sql)[0]
+  end
+
+  # Return details for a specific sample types.
+  #
+  # @param id [Int] the id of the sample type
+  #
+  # @return the sample types
+  def self.details(id)
+    # GET FEILD TYPES
+    sql = "
+      select *, null as 'allowable_field_types'
+      from field_types ft
+      where ft.parent_class = 'SampleType' and ft.parent_id = #{id}
+      order by ft.name
+    "
+    field_types = FieldType.find_by_sql sql
```

Comment:
Put queries into the appropriate classes as class methods.
This returns `FieldType` objects, so the method should be in the `FieldType` class

Reply:
Also, referencing the note above about ActiveRecord vs raw SQL, please document the rationale for any raw SQL queries with these methods

Reply:
raw sql queries are just faster.  I can change the simple ones, but Rails is horribly inefficient for complex relationships (I can show you many examples in the current code where Rails returns <n+1> queries (if not actually <x*n + 1>).

In general I prefer to just write the query directly (it is one less layer of abstraction), but I can change the basic ones.

Reply:
My point here is really about putting queries into methods in the class of the object returned about the query, and documenting the choice to use raw SQL. So, I'm just asking you to think specifically whether the raw SQL is an improvement.

Also, as we discussed, the performance hit is likely the conversion to ActiveRecord from the database result.